src/main/kotlin/com/velocitypowered/api/language/kotlin/ext/TabListEntryExt.kt

```diff
@@ -0,0 +1,63 @@
+package com.velocitypowered.api.language.kotlin.ext
+
+import com.velocitypowered.api.proxy.player.TabList
+import com.velocitypowered.api.proxy.player.TabListEntry
+import com.velocitypowered.api.util.GameProfile
+import net.kyori.adventure.text.Component
+
+val TabListEntry.parent: TabList
+    inline get() = parent()
+
+val TabListEntry.gameProfile: GameProfile
+    inline get() = gameProfile()
+
+var TabListEntry.displayName: Component?
+    inline get() = displayName()
+    inline set(value) {
+        setDisplayName(value)
+    }
+
+var TabListEntry.ping: Int
+    inline get() = ping()
+    inline set(value) {
+        setPing(value)
+    }
+
+var TabListEntry.gameMode: Int
+    inline get() = gameMode()
+    inline set(value) {
+        setGameMode(value)
+    }
+
+var TabListEntry.Builder.tabList: TabList
+    get() = throw UnsupportedOperationException("There is no getter in the Velocity API")
```

Comment:
You should be able to read a variable, otherwise it's not really... a variable...

Reply:
unfortunately the Velocity API does not provide getters for creating an ideomatic DSL builder on kotlin:

```kotlin
TabListEntry {
    latency = 50
    gameMode = 1
}
```

Reply:
It goes against the whole concept of variables to not be readable, too, though. This is a case where a method is a better choice, as that properly represents what's going on.